pkg/cloudprovider/aws/instancetype.go

```diff
@@ -369,12 +372,29 @@ func (i *InstanceType) computeMaxPods(ctx context.Context, kc *v1alpha5.KubeletC
 		mp = ptr.Int64(i.eniLimitedPods())
 	}
 
-	if kc != nil && ptr.Int32Value(kc.PodsPerCore) > 0 && amiFamily.PodsPerCoreEnabled() {
+	if kc != nil && ptr.Int32Value(kc.PodsPerCore) > 0 && i.amiFamily().PodsPerCoreEnabled() {
 		mp = ptr.Int64(lo.Min([]int64{int64(ptr.Int32Value(kc.PodsPerCore)) * ptr.Int64Value(i.VCpuInfo.DefaultVCpus), ptr.Int64Value(mp)}))
 	}
 	return mp
 }
 
+func (i *InstanceType) amiFamily() amifamily.AMIFamily {
+	return amifamily.GetAMIFamily(i.provider.AMIFamily, &amifamily.Options{})
+}
+
 func lowerKabobCase(s string) string {
 	return strings.ToLower(strings.ReplaceAll(s, " ", "-"))
 }
+
+func parsePercentage(v string) float64 {
```

Comment:
can you use mustParse to signify a panic to the caller is possible:
```suggestion
func mustParsePercentage(v string) float64 {
```